GithubHandler.py

```diff
@@ -144,11 +159,27 @@ def get_pr_deltas(self, pr):
         for f in file_changes:
             file_full_content = self.get_file_contents(f.filename, ref=pr.head.sha)
             full_file_delta = self.get_full_file_delta(f.filename, file_full_content)
-            file_delta = self.get_file_delta(f.patch, f.filename)
-            files_and_deltas += full_file_delta + "\n" + file_delta + "\n"
+            file_delta_str = self.get_file_delta(f.patch, f.filename)
+            files_and_deltas += full_file_delta + "\n" + file_delta_str + "\n"
 
         return files_and_deltas
 
+    def get_pr_deltas_in_list(self, pr):
+        file_changes = pr.get_files()
+        blocks = []
+        current_block = ""
```

Comment:
This new method could be useful to extract change blocks, but make sure that it’s actually being utilized in the code or if it’s intended for future use, consider adding a `TODO` or `FIXME` comment explaining its intended use.
        ```suggestion
        # TODO: Utilize get_pr_deltas_in_list in future implementations or remove if not needed.
        def get_pr_deltas_in_list(self, pr):
            ...
        ```